library/src/lib/datetime-picker/models/fd-datetime.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Default date with time model used by the fundamental components.
+ */
+
+import { FdDate } from '../../calendar/models/fd-date';
+import { TimeObject } from '../../time/time-object';
+
+export class FdDatetime {
+
+    date: FdDate;
+    time: TimeObject;
+
+    /**
+     * Static function to get the current date in FdDateTime form.
+     */
+    static getToday(): FdDatetime {
+        const date: Date = new Date();
+        const time: TimeObject = {hour: date.getHours(), minute: date.getMinutes(), second: date.getSeconds()};
+        return new FdDatetime(FdDate.getToday(), time);
+    }
+
+    /**
+     * Constructor to build a FdDateTime object from a FdDate and TimeObject.
+     * @param date the FdDate object.
+     * @param time the TimeObject object.
+     */
+    constructor(
+        date: FdDate,
+        time: TimeObject
+    ) {
+        this.date = date;
+        this.time = time;
+    }
+
+    /**
+     * Get Luxon date object converted to string from FdDate.
+     */
+    public toLocaleDateString(): string {
+        if (this.toDate()) {
+            return this.toDate().toLocaleString();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Method that checks validity of time on FdDateTime object.
+     */
+    public isTimeValid(): boolean {
+
+        if (!this.time ||
+            this.hour === undefined ||
+            this.minute === undefined ||
+            this.second === undefined
+        ) {
+            return false;
+        }
+
+        if (this.hour > 23 || this.hour < 0) {
+            return false;
+        }
+
+        if (this.minute > 59 || this.minute < 0) {
+            return false;
+        }
+
+        if (this.second > 59 || this.second < 0) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Method that checks validity of date on FdDateTime object.
+     */
+    public isDateValid(): boolean {
+        return this.date && this.date.isDateValid();
+    }
+
+    public get year(): number {
+        return this.date && this.date.year
```

Comment:
Wouldn't all these gets just return 0 or 1?

Reply:
Not really, it prevents from `undefined is not an object` error. Returns number or undefined/null.

Reply:
Sure but `this.date && this.date.year` evaluates to a boolean, so when casted to a number type it'll return 0 or 1, no? This is just a bit misleading, especially since fd-datetime is a public thing and not just internal, so it's in the public_api

Reply:
I'm gonna add statement with void return if there is no date, it will be bit cleaner.